### PR TITLE
better return type for fns that returns JWTResponse

### PR DESCRIPTION
**Pull request by nirgur**
fixes: https://github.com/descope/etc/issues/859

use the `wrapWith` utility from core-js to return correct types 
image

### CODE REVIEW

File: lib/helpers.ts [hunk 1]
```diff
@@ -128,3 +77,20 @@ export function getAuthorizationClaimItems(
     : authInfo.token[claim];
   return Array.isArray(value) ? value : [];
 }
+
+/**
+ * Add hooks to an existing core-sdk config
+ */
+export const addHooks = <Config extends CoreSdkConfig>(
```
Comment (asafshen) [inline lib/helpers.ts:84]: this code was duplicated from web-js-sdk

might worth putting it in core-sdk with the rest of the wrapwith there

**Review by asafshen — APPROVED** [review]
👍 
this issue was opened because the examples didn't work properly, can we make sure now they are working well? (or maybe other things need to be fixed, can be done in another PR ofc)

**Comment by nirgur** [comment]
> 👍 this issue was opened because the examples didn't work properly, can we make sure now they are working well? (or maybe other things need to be fixed, can be done in another PR ofc)

The code was changes since this issue was opened
Looks like there are things that should be fixed there but I don't see errors about those types

File: lib/helpers.ts (hunk 1 above)
Comment (nirgur) [reply lib/helpers.ts:84]: https://github.com/descope/core-js-sdk/pull/147

Comment (nirgur) [reply lib/helpers.ts]: done

**Review by asafshen — APPROVED** [review]

**Pull request merged**